storm-server/src/main/java/org/apache/storm/scheduler/resource/strategies/scheduling/BaseResourceAwareStrategy.java

```diff
@@ -148,9 +153,12 @@ protected WorkerSlot findWorkerForExec(ExecutorDetails exec, TopologyDetails td,
         for (String id : sortedNodes) {
             RAS_Node node = nodes.getNodeById(id);
             if (node.couldEverFit(exec, td)) {
+                Collection<WorkerSlot> topologyUsedSlots = oneWorkerPerExecutor ? node.getUsedSlots(td.getId()) : null;
```

Comment:
Set this to `Collections.emptySet()` instead of null, then you can get rid of the null check below.